src/Main.elm

```diff
@@ -1,10 +1,418 @@
 module Main exposing (main)
 
-import Data.Audience
-import Data.AudienceFolder
-import Html exposing (Html)
+import Browser
+import Data.Api as Api
+import Data.Audience as Audience exposing (Audience, AudienceType)
+import Data.AudienceFolder exposing (AudienceFolder)
+import Data.Store as Store exposing (AudienceFolderID, AudienceLevel, Store)
+import Element exposing (Element, column, el, none, paragraph, row, text)
+import Element.Background
+import Element.Border
+import Element.Events
+import Element.Font
+import Element.Input exposing (button)
+import Html
+import Html.Attributes
+import Http
+import Task
 
 
-main : Html msg
+
+---------------
+-- M O D E L --
+---------------
+
+
+type alias InitialisingState =
+    { folders : Maybe (List AudienceFolder)
+    , audiences : Maybe (List Audience)
+    }
```

Comment:
this type imo shows one of the weaknesses of tea.. things like this has to be dealth with using custom state handling. Aff monad would be probably much nicer for modeling this :/